examples/keras_recipes/tensorflow_numpy_models.py

```diff
@@ -0,0 +1,340 @@
+    model.add(DenseTNP(3))
+    # The model will have no issue using a normal Dense layer
+    model.add(Dense(3, use_bias=True))
+    # ... or switching back to tnp layers!
+    model.add(LinearProjector())
+    return model
+
+
+model = create_mixed_model()
+model.compile(
+    optimizer="adam",
+    loss="mean_squared_error",
+    metrics=[keras.metrics.MeanAbsolutePercentageError()],
+)
+model.build((None, 13,))
+model.summary()
+
+evaluate_model(model)
+
+"""
+The Keras API offers a wide variety of layers.  The ability to use them alongside NumPy
+code can be a huge time saver in projects.
+"""
+
+"""
+# Distribution Strategy
+Tensorflow NumPy and Keras integrate with [Tensorflow distribution
+strategies](https://www.tensorflow.org/guide/distributed_training).  This makes it simple
+to perform distributed training across multiple GPUs, or even an entire TPU Pod.
+
+"""
+
+gpus = tf.config.list_logical_devices("GPU")
+if gpus:
+    strategy = tf.distribute.MirroredStrategy(gpus)
+else:
+    # We can fallback to a no-op CPU strategy.
+    strategy = tf.distribute.get_strategy()
+print("Running with strategy:", str(strategy.__class__.__name__))
+
+with strategy.scope():
+    model = create_layered_tnp_model()
+    model.compile(
+        optimizer="adam",
+        loss="mean_squared_error",
+        metrics=[keras.metrics.MeanAbsolutePercentageError()],
+    )
+    model.build((None, 13,))
+    model.summary()
+    evaluate_model(model)
+
+"""
+# Tensorboard Integration
+One of the many benefits of using the Keras API is the ability to monitor training
+through Tensorboard.  Using the Tensorflow NumPy API alongside Keras allows you to easily
+leverage Tensorboard.
+"""
+
+keras.backend.clear_session()
+
+"""
+To load the Tensorboard from a Jupyter notebook, you can run the following magic:
+```
+%load_ext tensorboard
+```
+
+"""
+
+models = [
+    (TNPForwardFeedRegressionNetwork(blocks=[3, 3]), "TNPForwardFeedRegressionNetwork"),
+    (create_layered_tnp_model(), "layered_tnp_model"),
+    (create_mixed_model(), "mixed_model"),
+]
+for model, model_name in models:
+    model.compile(
+        optimizer="adam",
+        loss="mean_squared_error",
+        metrics=[keras.metrics.MeanAbsolutePercentageError()],
+    )
+    model.fit(
+        x_train,
+        y_train,
+        epochs=200,
+        verbose=0,
+        callbacks=[keras.callbacks.TensorBoard(log_dir=f"logs/{model_name}")],
+    )
+
+"""
+To load the Tensorboard from a Jupyter notebook you can use the `%tensorboard` magic:
+```
+%tensorboard --logdir logs
+```
+"""
+
+"""
+The Tensorboard monitor metrics and examine the training curve.
+"""
+
+"""
+![Tensorboard training graph](https://i.imgur.com/wsOuFnz.png)
+"""
+
+"""
+The Tensorboard also allows you to explore the computation graph used in your models.
+"""
+
+"""
+![Tensorboard graph exploration](https://i.imgur.com/tOrezDL.png)
+"""
+
+"""
+  The ability to introspect into your models can be valuable during debugging.
+"""
```

Comment:
You can use a single text block for everything above.

Reply:
Done!